src/routes/credentials.rs

```diff
@@ -282,23 +295,78 @@ async fn post_credential_handler(
         .get(&oidc_context.client_id)
         .ok_or(Error::OauthInvalidClientId)?;
     if let Some(checks_directory) = &client_config.checks_directory {
-        let checker = app_state
+        let mut app_state_write = app_state.write().await;
+        let checker = app_state_write
             .rhai_checkers
             .get_or_create(&oidc_context.client_id, checks_directory)?;
         checker.check(&id_token)?;
+        drop(app_state_write);
+    }
+
+    let response_type = session
+        .get::<String>(RESPONSE_TYPE_SESSION_KEY)?
+        .ok_or(Error::ResponseType)?;
+
+    drop(app_state_read);
+
+    if response_type == "code" {
+        log::info!("Authorization Code Flow");
+        let code = generate_random_string();
+
+        // Store (code -> token_response) so it can be sent later at the `/token` endpoint.
+        let token_response = TokenResponse {
+            token_type: "bearer".to_string(),
+            access_token: generate_random_string(),
```

Comment:
`access_token` is required, although it has no use because the Authorization Server doesn't provide any additional functionalities that can make use of it, hence it's not stored.

Reply:
Except, it is stored, no? We store the whole `token_response` structure.

Reply:
Ok so it's stored but not used.